lib/runtime/types.go

```diff
@@ -81,14 +88,146 @@ type Context struct {
 	NodeStorage NodeStorage
 	Network     BasicNetwork
 	Transaction TransactionState
-
+	SigVerifier *SignatureVerifier
 	// TransactionStorageChanges is used by ext_storage_start_transaction to keep track of
 	// changes made after it's called. The next call to ext_storage_commit_transaction will
 	// commit all the changes, or if ext_storage_rollback_transaction is called, the changes
 	// will be discarded.
 	TransactionStorageChanges []*TransactionStorageChange
 }
 
+// Signature ...
+type Signature struct {
```

Comment:
maybe you could move all the signature related code to a new file, since there's quite a lot of code